### PR TITLE
Add more static tooling

### PR DESCRIPTION
**Pull request by SimonRichardson**
##  Description of change

The following adds more static tooling to the shell script tests,
with this change we've got parity with the previous .git/pre-check
tests.

## QA steps

```sh
./main.sh static_analysis test_static_analysis_go
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/WKupgG1G/1873-add-more-static-tooling

File: tests/suites/static_analysis/lint_go.sh [hunk 1]
```diff
@@ -108,14 +131,19 @@ test_static_analysis_go() {
...
 
     ## golint
     if which golint >/dev/null 2>&1; then
       run "run_go_lint"
     fi
 
+    ## goimports
+    if which goimports >/dev/null 2>&1; then
+      run "run_go_imports" "${FOLDERS}"
+    fi
```
Comment (hpidcock) [inline tests/suites/static_analysis/lint_go.sh]: It would be nice if these had output to mention they are being skipped.

Comment (SimonRichardson) [reply tests/suites/static_analysis/lint_go.sh]: They do, the run command outputs if they're skipped 😉

Comment (SimonRichardson) [reply tests/suites/static_analysis/lint_go.sh]: Yeah, sorry, I get what you mean - yes, I'll change this in a future PR. I did it so it was optional at first, but we've gone all in, so we should just be explicit.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

    Windows failure

**Pull request merged**